### PR TITLE
📏 #256 Encryption Device Schema

### LINKED ISSUE #256: ✨  Data Schema for Encryption 1.0.0 - Preview 1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

Create a data schema in preparation for release of 1.0.0 . This will be part of three releases (`1.0.0-preview-1`, `1.0.0-preview-2`, `1.0.0`.

This will includes data schema and sample data for all the inputs and outputs in Encryption for ElectionGuard. 

Input:
- `PlaintextBallot`
- `Manifest`
- `ElectionConstants`
- `CiphertextElectionContext`

Output:
- `SubmittedBallot`
- `EncryptionDevice`

     
 Sample data can be created at this juncture to align with these. These should be generated from Python based on the current date. 

### Possible Implementation

_No response_

### Anything else?

- [x] `PlaintextBallot` created manually
- [x] `Manifest` should be very similar to the current one in 0.95.0
- [x] `ElectionConstants`, `CiphertextElectionContext`, `EncryptionDevice` can be autocreated in python
- [x] `SubmittedBallot` is available on slack
- [x] Rename Data Format to Data Schema on Data Format page. Ensure links work.
- [ ] Upload sample data for `1.0.0-preview-1`

See here:
https://github.com/microsoft/electionguard/tree/main/data/0.95.0

See here:
https://github.com/microsoft/electionguard/tree/main/data/0.95.0
https://github.com/microsoft/electionguard/blob/main/docs/develop/Data_Format.md

### PR DESCRIPTION
**Pull request by lprichar**
### Issue

Fixes #256

### Description
Updates the schemas to the latest version in the Python repo.  Also renames data format page to data schema.

### Testing
1. Go to the [Data Schema](https://github.com/microsoft/electionguard/blob/main/docs/develop/Data_Format.md) page and click the links to ensure they all work.
2. Validate the schemas with a tool like [JSON Schema Validator](https://www.jsonschemavalidator.net/)

### CODE REVIEW

File: data/0.95.0/schema/ciphertext_election_context.schema.json [hunk 1]
```diff
@@ -1,31 +1,22 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema",
```
Comment (keithrfung) [inline data/0.95.0/schema/ciphertext_election_context.schema.json:2]: These should be in their own folder `data/1.0.0-preview-1/schema` with their own sample data.

File: data/0.95.0/schema/plaintext_ballot.schema.json [hunk 2]
```diff
@@ -1,4 +1,83 @@
...
+          "object_id": {
+            "title": "Object Id",
+            "type": "string"
+          },
+          "sequence_order": {
+            "title": "Sequence Order",
+            "description": "Used for ordering in a ballot to ensure various encryption primitives are deterministic",
+            "type": "integer"
+          },
+          "ballot_selections": {
```
Comment (keithrfung) [inline data/0.95.0/schema/plaintext_ballot.schema.json:34]: These should be using `definitions`. As should the contests.

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-mkdocs==1.2.3
+mkdocs==1.2.4
```
Comment (keithrfung) [inline requirements.txt:1]: Maybe should be in its own PR, but gonna let this slide in for brevity.

File: data/0.95.0/schema/election_constants.schema.json [hunk 4]
```diff
@@ -1,5 +1,4 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema",
```
Comment (keithrfung) [inline data/0.95.0/schema/election_constants.schema.json:2]: We want to define the schema in ALL of these files. The schema shifts frequently and it can check the json format.

File: data/0.95.0/schema/ciphertext_election_context.schema.json [hunk 5]
```diff
@@ -40,26 +31,34 @@
...
         },
         "crypto_base_hash": {
-          "$ref": "#/definitions/ElementModQ"
+          "title": "Crypto Base Hash",
+          "type": "string"
         },
         "crypto_extended_base_hash": {
-          "$ref": "#/definitions/ElementModQ"
+          "title": "Crypto Extended Base Hash",
+          "type": "string"
```
Comment (keithrfung) [inline data/0.95.0/schema/ciphertext_election_context.schema.json:51]: These should still reference ElementModP and ElementModQ . See the Submitted Ballot format.

**Review by keithrfung — CHANGES_REQUESTED** [review]

File: data/0.95.0/schema/ciphertext_election_context.schema.json (hunk 5 above)
Comment (lprichar) [reply data/0.95.0/schema/ciphertext_election_context.schema.json:51]: This file was purely code generated per the instructions in the ticket.  As long as I'm customizing it, I'll put in some descriptions.

File: data/1.0.0-preview-1/schema/lagrange_coefficients_record.schema.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "title": "ParsingModel[LagrangeCoefficientsRecord]",
+  "$ref": "#/definitions/LagrangeCoefficientsRecord",
+  "definitions": {
+    "ElementModQ": {
+      "title": "ElementModQ",
```
Comment (keithrfung) [inline data/1.0.0-preview-1/schema/lagrange_coefficients_record.schema.json:7]: This is actually wrong. Data is no longer included. Check the other files to ensure this mistake isn't repeated.
```
"ElementModQ": {
      "title": "ElementModQ"
       "type": "string"
       }
```

**Review by keithrfung — CHANGES_REQUESTED** [review]

File: data/1.0.0-preview-1/schema/lagrange_coefficients_record.schema.json (hunk 6 above)
Comment (lprichar) [reply data/1.0.0-preview-1/schema/lagrange_coefficients_record.schema.json:7]: Nice catch!  It actually wasn't part of the ticket, I just copied it over from the .95 spec.  There are no other files like that.

**Review by keithrfung — APPROVED** [review]

**Comment by shreyasminocha** [comment]
In commit 9dfbf0a460c93c9dd15d1146c08c2edaf5f4f1fd the constants for the hamilton-general election were changed from the parameters in the spec to much smaller values. Was this change intentional? If so, could you please also include sample data that uses the standard parameters?

**Comment by colleenirt** [comment]
Hi @shreyasminocha that sample is using test constants.  We have a story to enhance the Python CLI to be able to generate sample data.  https://github.com/microsoft/electionguard-python/issues/618
I don't yet know when that will be worked.
